confgenerator/confgenerator.go

```diff
@@ -209,35 +211,49 @@ func generateOtelServices(receiverNameMap map[string]string, exporterNameMap map
 		if strings.HasPrefix(pID, "lib:") {
 			return nil, fmt.Errorf(`pipeline id prefix 'lib:' is reserved for pre-defined pipelines. Pipeline ID %q is not allowed.`, pID)
 		}
+
 		p := pipelines[pID]
+		// if the pipeline is default_pipeline, then split the pipeline into different streams.
+		if pID == "default_pipeline" {
```

Comment:
Do we have to restrict this to pID == "default_pipeline"? If users name the pipeline to some other names, what happened? I wonder if we can remove this if check.

Reply:
because we try to hide the underlying processors for hostmetrics, mssql and iis and somehow squeenze those three into one pipeline. Underneath it, inside the OTEL config, those are three pipeline. That's why this default_pipeline is so unique and it should be reserved for this usage.

Reply:
We should have the same behavior for any pipeline, not just the default pipeline.